src/resolve/resolved-dnstls-openssl.c

```diff
@@ -87,6 +87,11 @@ int dnstls_stream_connect_tls(DnsStream *stream, DnsServer *server) {
                         return -ECONNREFUSED;
         }
 
+        if (server->server_name)
+                r = SSL_set_tlsext_host_name(s, server->server_name);
+                if (r <= 0)
+                        return r;
```

Comment:
Missing `{}`.
```
if (server->server_name) {
    r = SSL_set_tlsext_host_name(...);
    if (r <= 0) {
        char errbuf[256];

        error = ERR_get_error();
        ERR_error_string_n(error, errbuf, sizeof(errbuf));
        log_debug("Failed to set server name: %s", buf);
        return -EINVAL;
    }
}
```

Reply:
Thanks, That was exactly what I was going to ask... I was quite lost with error codes.

I see that `SSL_set_session` for example isn't checked.

Reply:
Yeah... Bad. But in this case, checking error is important, as user may input invalid server name.